### ISSUE TITLE
"Follow" for a polygon

### ISSUE THREAD

**Issue by Mahabarata** [openstreetmap/iD#9022]
### URL

Any

### How to reproduce the issue?

Hi,

If I want to create a new way beginning on a node of another feature (linear way or polygon) and click on F (to follow), I have a message "This feature can't follow another feature because it isn't connected to enough consecutive points along another feature. Add another point manually to continue." : it's ok because ID can't know if I want to follow the feature in one direction or the other.

But if I want to create a new polygon rather than a way and click on F, I have no message and a node is created. The strangest thing is that the new added point don't follow the direction of the other feature but goes backward (if you change the direction of the other feature before to create the new polygon, you will see that).

Why not to do the same as with ways and just add an error message ?

It seems that in transifex we have the appropriate message : "This feature can't follow the area because it is only connected at a single point. Add another point manually to continue."

Best regards

### Which iD Editor versions do you see the issue on?

Released version at openstreetmap.org/edit

### Which browsers are you seeing this problem on?

Chrome

**Comment by danieldegroot2** [comment]
Related #8750.

**Comment by tyrasd** [comment]
good catch. the bug originates from the fact that areas start with two nodes (first node == last node). This is not properly checked by the follow algorithm.
